analyzers/src/SonarAnalyzer.CSharp/Extensions/AssignmentExpressionSyntaxExtensions.cs

```diff
@@ -108,17 +120,15 @@ private static bool MapTupleElements(ImmutableArray<AssignmentMapping>.Builder a
             {
                 var leftExpression = leftEnumerator.Current.Expression;
                 var rightExpression = rightEnumerator.Current.Expression;
-                if (TupleExpressionSyntaxWrapper.IsInstance(leftExpression)
-                    && TupleExpressionSyntaxWrapper.IsInstance(rightExpression))
+                switch (HandleTupleNesting(arrayBuilder, leftExpression, rightExpression))
                 {
-                    if (!MapTupleElements(arrayBuilder, (TupleExpressionSyntaxWrapper)leftExpression, (TupleExpressionSyntaxWrapper)rightExpression))
-                    {
+                    case true:  // Nesting was handled.
```

Comment:
the the status of the nesting match, as discussed it might be better to use an enum, smth like
```
enum NestingMatch
{
  Handled,
  Failed,
  Leaf
}
```